### ISSUE TITLE
remove autoconf temporary files from repo

### ISSUE THREAD

**Issue by ssinyagin** [irontec/sngrep#20]
after "./configure && make", a number of files got modified by automake:

```
#       modified:   Makefile.in
#       modified:   aclocal.m4
#       modified:   config/Makefile.in
#       modified:   configure
#       modified:   doc/Makefile.in
#       modified:   src/Makefile.in
```
It's better to remove all autogenerated files completely from the repository, and run "autoreconf" after cloning.

here's an example of .gitignore:
https://github.com/ssinyagin/spsid-server/blob/master/.gitignore

**Comment by Kaian** [comment]
This is something about autotools that makes me really mad.

Usually, most software that use autotools has a general **configure** script that does not require autoreconf to be executed, but I found some problems where aclocal or automake versions where specified in configure script so, if the installed versions are different autoreconf is required (the make process runs it automagically), what creates new versions of scripts (and those differences in repository status).

I considered removing the templates and add that _extra_ command in build instructions, but I wonder if there is a way to create a really generic configure that does not require autoreconf at all.

tl;dr: Why configure script requires autoreconf :(

**Comment by Kaian** [comment]
What would you think about a mixed solution?

Include generated scripts and .in templates in the repo, but ignore changes in them. 

The only reason I see them usefull is that most people will expect ./configure script to be there, without reading INSTALL instructions or wiki pages. 

Anyway I agree autoreconf is the right thing to be done in all cases, but not something most users do _naturally_ (although make will do it for them).

**Comment by ssinyagin** [comment]
usually it works fine if I tell them to use autoreconf after cloning. My users are trained :) 

You can still do "make dist" and put the resulting .tar.gz somewhere for downloading -- it will contain all required scripts. And keep the git repo for those who are familiar with managing the sources :)

btw, a debian package will be great :)

**Comment by Kaian** [comment]
Actually what I have suggested doesn't seem to be possible :|
Even adding files to .gitignore,  if they are in the repo, changes will be shown.

About the debian package, thats something I'm fighting but I haven't found a sponsor yet :(
http://mentors.debian.net/package/sngrep

In the initial versions, a debian directory was also included in the repo, but some users suggested me to remove it (see 41a2fab5466d84cd778ffc6585ffeca2f0f1c80f)

**Comment by ssinyagin** [comment]
so, "make dist" and upload somewhere is probably the easiest way -- you can simply put a link into README.

what does it mean to be a sponsor for a deb package? I'm not familiar with their rules, I'm just a user :)

**Comment by Kaian** [comment]
The debian sponsor is the one that uploads a package to their repositories, while the maintainer creates new versions and handle software issues.

I have created a branch without the generated files, some ignores and a new script (bootstrap.sh) as autotools documentation suggests. Also, updated install instructions in README file.

Generating a .deb won't be a problem, but! I should create for more than one arch and debian distribution. For example, squeeze version should link binary agains libssl-0.9.8 and jessie one should use 1.0.0.

**Comment by ssinyagin** [comment]
you can also create your own repository, like freeswitch is doing :)
Are you on skype? There are more than 20 people with the same name as
yours. Mine is ssinyagin.

**Comment by Kaian** [comment]
@irontec has internal debian repositories to distribute its software, but the need for multiple binaries for each distribution (and multiple repositories) will be required anyway.

I only have personal Skype account :angel: but you can find me anytime at irc.freenode.net as Kaian :smile:

**Comment by ssinyagin** [comment]
your patch worked perfectly on debian 7 (obviously). I'll bring up a debian 6 machine in few minutes

**Comment by Kaian** [comment]
Nice!! Thanks a lot for the testing!

I'm also tested successfully with gentoo and centos, so I have update the 0.2.2 release :)

**Comment by ssinyagin** [comment]
worked fine in debian 6. Here are the pre-requisite installation instructions for your README:
```
apt-get update && \
apt-get install -y git autoconf automake gcc make \
libncurses-dev libpcap-dev libssl-dev
```

**Comment by Kaian** [comment]
Updated wiki instructions for debian https://github.com/irontec/sngrep/wiki/Building
